src/app/clusters/scenes-server/SceneTable.h

```diff
@@ -127,148 +127,137 @@ class SceneHandler : public IntrusiveListNodeBase<>
+        mExtensionFieldSets = other.mExtensionFieldSets;
+    }
+    ~SceneData(){};
 
-        void SetName(const CharSpan & sceneName)
-        {
-            if (nullptr == sceneName.data())
-            {
-                mName[0]    = 0;
-                mNameLength = 0;
-            }
-            else
-            {
-                size_t maxChars = std::min(sceneName.size(), kSceneNameMaxLength);
-                memcpy(mName, sceneName.data(), maxChars);
-                mNameLength = maxChars;
-            }
-        }
+    bool operator==(const SceneData & other) const
+    {
+        return ((CharSpan(mName, mNameLength).data_equal(CharSpan(other.mName, other.mNameLength))) &&
+                (mSceneTransitionTimeMs == other.mSceneTransitionTimeMs) && (mExtensionFieldSets == other.mExtensionFieldSets));
+    }
 
-        void Clear()
+    void SetName(const CharSpan & sceneName)
+    {
+        if (nullptr == sceneName.data())
         {
-            SetName(CharSpan());
-            mSceneTransitionTimeMs = 0;
-            mExtensionFieldSets.Clear();
+            mName[0]    = 0;
+            mNameLength = 0;
         }
-
-        void operator=(const SceneData & other)
+        else
         {
-            SetName(CharSpan(other.mName, other.mNameLength));
-            mExtensionFieldSets    = other.mExtensionFieldSets;
-            mSceneTransitionTimeMs = other.mSceneTransitionTimeMs;
+            size_t maxChars = std::min(sceneName.size(), kSceneNameMaxLength);
+            memcpy(mName, sceneName.data(), maxChars);
+            mNameLength = maxChars;
         }
-    };
+    }
 
-    /// @brief Struct combining both ID and data of a table entry
-    struct SceneTableEntry
+    void Clear()
     {
-        // ID
-        SceneStorageId mStorageId;
-
-        // DATA
-        SceneData mStorageData;
+        SetName(CharSpan());
+        mSceneTransitionTimeMs = 0;
+        mExtensionFieldSets.Clear();
+    }
 
-        SceneTableEntry() = default;
-        SceneTableEntry(SceneStorageId id) : mStorageId(id) {}
-        SceneTableEntry(const SceneStorageId id, const SceneData data) : mStorageId(id), mStorageData(data) {}
-
-        bool operator==(const SceneTableEntry & other) const
-        {
-            return (mStorageId == other.mStorageId && mStorageData == other.mStorageData);
-        }
+    void operator=(const SceneData & other)
+    {
+        SetName(CharSpan(other.mName, other.mNameLength));
+        mExtensionFieldSets    = other.mExtensionFieldSets;
+        mSceneTransitionTimeMs = other.mSceneTransitionTimeMs;
+    }
+};
+} // namespace scene_table_elements
 
-        void operator=(const SceneTableEntry & other)
-        {
-            mStorageId   = other.mStorageId;
-            mStorageData = other.mStorageData;
-        }
-    };
+template <class EFStype>
+class SceneTable
+    : public virtual app::common::FabricTable<scene_table_elements::SceneStorageId, scene_table_elements::SceneData<EFStype>>
```

Comment:
Why do we need virtual inheritance here?

Reply:
No longer need it with `FabricTable` eliminated